### PR TITLE
Feature branch refactor

### PR DESCRIPTION
**Pull request by ivar-lazzaro**

### CODE REVIEW

File: opflexagent/gbp_ovs_agent.py [hunk 1]
```diff
@@ -10,365 +10,187 @@
...
-            "configured" if self.from_config else "auto-allocated"))
+class GBPOpflexAgent(sg_rpc.SecurityGroupAgentRpcCallbackMixin,
+                     rpc.GbpNeutronAgentRpcCallbackMixin):
+    """Group Based Policy Opflex Agent.
 
-    def is_valid(self):
-        return ((self.ip_start and self.ip_gateway) or
-                (self.ip6_start and self.ip6_gateway))
+    The GBP opflex agent assumes a pre-existing bridge (integration bridge is
+    not required). This agent is an interface between the SDN controller and
```
Comment (amittbose) [inline opflexagent/gbp_ovs_agent.py]: The second sentence reads wrong. How about:

This agent is an interface between Openstack Neutron and agent-ovs, which renders policies from an OpFlex-based SDN controller (like Cisco ACI) into OpenFlow rules for OVS.

File: opflexagent/gbp_ovs_agent.py [hunk 2]
```diff
@@ -10,365 +10,187 @@
...
+        self.run_daemon_loop = True
+        # The initialization is complete; we can start receiving messages
+        self.connection.consume_in_threads()
+
+        self.quitting_rpc_timeout = kwargs['quitting_rpc_timeout']
+        # Initialize the Endpoint Manager.
+        # TODO(ivar): make this component pluggable.
+        self.ep_manager = ep_manager.EndpointFileManager().initialize(
+            self.host, self.bridge_manager, kwargs)
+        # Clenup SNAT rules.
```
Comment (amittbose) [inline opflexagent/gbp_ovs_agent.py]: Is this a stray comment?

File: opflexagent/test/test_endpoint_file_manager.py [hunk 3]
```diff
@@ -0,0 +1,569 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
```
Comment (amittbose) [inline opflexagent/test/test_endpoint_file_manager.py]: This seems to have a lot of the same test code as test_gbp_ovs_agent.py. Perhaps you can add this file later when you are ready to refactor test_gbp_ovs_agent.py? Maintaining 2 copies will be a pain.

File: opflexagent/gbp_ovs_agent.py (hunk 1 above)
Comment (ivar-lazzaro) [inline opflexagent/gbp_ovs_agent.py]: Done

File: opflexagent/gbp_ovs_agent.py (hunk 2 above)
Comment (ivar-lazzaro) [inline opflexagent/gbp_ovs_agent.py]: yes, removing

File: opflexagent/test/test_endpoint_file_manager.py (hunk 3 above)
Comment (ivar-lazzaro) [inline opflexagent/test/test_endpoint_file_manager.py]: ok makes sense to me

**Comment by ivar-lazzaro** [comment]
Addressed by https://github.com/noironetworks/python-opflex-agent/pull/24

**Pull request closed without merging**